lib/presentation/signup/bloc/sign_up_bloc.dart

```diff
@@ -0,0 +1,205 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:get/get.dart';
+
+import '../../../data/constants/routes.dart';
+import '../../../data/constants/strings.dart';
+import '../../../data/services/local/image_service.dart';
+import '../../../domain/models/handle.dart';
+import '../../../domain/models/sign_up.dart';
+import '../../../domain/models/status.dart';
+import '../../../domain/repositories/user_repository.dart';
+import '../../../utils/failures.dart';
+import '../widgets/signup_widgets.dart';
+
+part 'sign_up_bloc.freezed.dart';
+part 'sign_up_event.dart';
+part 'sign_up_state.dart';
+
+/// The Bloc for the sign-up screen.
+class SignUpBloc extends Bloc<SignUpEvent, SignUpState> {
+  /// The Bloc for the sign-up screen.
+  SignUpBloc() : super(const SignUpState()) {
+    on<Initialize>(_initialize);
+
+    on<Back>(_handleBack);
+    on<Next>(_handleNext);
+
+    on<EmailInput>(_updateEmail);
+    on<InstituteInput>(_updateInstitute);
+    on<NameInput>(_updateName);
+    on<PasswordInput>(_updatePassword);
+    on<PlatformHandleInput>(_updateHandles);
+    on<UsernameInput>(_updateUsername);
+
+    on<SelectImage>(_selectImage);
+    on<ToggleObscure>(_toggleObscurePassword);
+  }
+
+  Future<void> _initialize(Initialize event, Emitter<SignUpState> emit) async {
+    // initialize controllers
+    final newState = state.copyWith(
+      emailController: TextEditingController(),
+      instituteController: TextEditingController(),
+      nameController: TextEditingController(),
+      passwordController: TextEditingController(),
+      usernameController: TextEditingController(),
+      handleControllers: {
+        'codechef': TextEditingController(),
+        'hackerrank': TextEditingController(),
+        'codeforces': TextEditingController(),
+        'spoj': TextEditingController(),
+      },
```

Comment:
No LeetCode while asking for the platform handles? 

CC: @akanksha-v

Reply:
Fixed in bc9a94c.